x-pack/plugins/data_enhanced/server/search/eql_search_strategy.ts

```diff
@@ -21,19 +21,18 @@ export const eqlSearchStrategyProvider = (
   logger: Logger
 ): ISearchStrategy<EqlSearchStrategyRequest, EqlSearchStrategyResponse> => {
   return {
-    cancel: async (context, id) => {
+    cancel: async ({ esClient }, id) => {
       logger.debug(`_eql/delete ${id}`);
-      await context.core.elasticsearch.client.asCurrentUser.eql.delete({
+      await esClient.asCurrentUser.eql.delete({
         id,
       });
     },
-    search: (request, options, context) =>
+    search: ({ esClient, uiSettingsClient }, request, options) =>
```

Comment:
I think we should change order of the arguments 
`request, options, { esClient, uiSettingsClient }`
in the future, we still need to unify the code between client and server and create a common interface for this. This small change allows us to temporarily mark the third argument as optional.